crypto/chacha/asm/chacha-x86_64.pl

```diff
@@ -107,24 +112,16 @@
 .asciz	"ChaCha20 for x86_64, CRYPTOGAMS by <appro\@openssl.org>"
 ___
 
-sub AUTOLOAD()          # thunk [simplified] 32-bit style perlasm
-{ my $opcode = $AUTOLOAD; $opcode =~ s/.*:://;
-  my $arg = pop;
-    $arg = "\$$arg" if ($arg*1 eq $arg);
-    $code .= "\t$opcode\t".join(',',$arg,reverse @_)."\n";
-}
-
-@x=("%eax","%ebx","%ecx","%edx",map("%r${_}d",(8..11)),
-    "%nox","%nox","%nox","%nox",map("%r${_}d",(12..15)));
-@t=("%esi","%edi");
+my @x=("%eax","%ebx","%ecx","%edx",map("%r${_}d",(8..11)),
+       "%nox","%nox","%nox","%nox",map("%r${_}d",(12..15)));
+my @t=("%esi","%edi");
 
 sub ROUND {			# critical path is 24 cycles per round
 my ($a0,$b0,$c0,$d0)=@_;
 my ($a1,$b1,$c1,$d1)=map(($_&~3)+(($_+1)&3),($a0,$b0,$c0,$d0));
 my ($a2,$b2,$c2,$d2)=map(($_&~3)+(($_+1)&3),($a1,$b1,$c1,$d1));
 my ($a3,$b3,$c3,$d3)=map(($_&~3)+(($_+1)&3),($a2,$b2,$c2,$d2));
-my ($xc,$xc_)=map("\"$_\"",@t);
-my @x=map("\"$_\"",@x);
+my ($xc,$xc_)=@t;
```

Comment:
> removing surprises like this is part of this.

I'd argue that if this is the case, then they are trying to understand wrong thing :-) They would be looking at perl when they should be looking at assembler :-) As already said perlasm is ultimately about assembler, perl is kind of expendable :-) You also have to recognize that there are things that can't be unseen.I mean yes, it might sting perl evangelist's eye, but would it keep assembler programmer from writing code? Please note that I'm *not* actually arguing that perlasm establishes some kind of styling standard, only that it was never the goal. This also means that I'm *not* saying that that change is out of question...